extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -848,24 +849,25 @@ private void contextChecks(QueryContext queryContext)
     }
 
     MSQSelectDestination selectDestination = MultiStageQueryContext.getSelectDestination(queryContext);
-    if (selectDestination == MSQSelectDestination.DURABLE_STORAGE) {
-      checkForDurableStorageConnectorImpl();
+    if (MSQSelectDestination.DURABLESTORAGE.equals(selectDestination)) {
+      checkForDurableStorageConnectorImpl(NodeRole.BROKER);
     }
   }
 
-  private void checkForDurableStorageConnectorImpl()
+  private void checkForDurableStorageConnectorImpl(NodeRole nodeRole)
```

Comment:
This parameter is always NodeRole.BROKER. Why is this change required?

Reply:
I initially envisioned to move the check out of this class but I decided against it. Hence this is left over code.